components/fxa-client/ios/FxAClient/FirefoxAccount.swift

```diff
@@ -145,52 +90,52 @@ open class FirefoxAccount {
     /// to make that call. The caller should then start the OAuth Flow again with
     /// the "profile" scope.
     open func getProfile(completionHandler: @escaping (Profile?, Error?) -> Void) {
-        queue.async {
+        DispatchQueue.global().async {
             do {
-                let profileBuffer = try FirefoxAccountError.unwrap { err in
-                    fxa_profile(self.raw, false, err)
-                }
-                let msg = try! MsgTypes_Profile(serializedData: Data(rustBuffer: profileBuffer))
-                fxa_bytebuffer_free(profileBuffer)
-                let profile = Profile(msg: msg)
+                let profile = try self.getProfileSync()
                 DispatchQueue.main.async { completionHandler(profile, nil) }
-                self.tryPersistState()
             } catch {
                 DispatchQueue.main.async { completionHandler(nil, error) }
             }
         }
     }
 
+    open func getProfileSync() throws -> Profile {
```

Comment:
I grew these `...Sync` and kept the default async methods in order to keep this class backwards compatible so Lockwise iOS can still continue to use it.
The `...Sync` methods are used by the account manager (which runs on its own queue).

Reply:
That was a good idea that I took a bit further in order to clean things up and make the code (I believe) way more readable. I ended up with 3 classes:
- `RustFxAccount`: only does Rust calls and Protobuf types conversions. That's it. No more persistence callbacks etc.
- `FxAccount: RustFxAccount`: adds persistence call backs + auth problems signaling to the manager.
- `FirefoxAccount: RustFxAccount`: async calls on top of RustFxAccount. Keep backwards compatibility. Marked as deprecated.